lib/maths/CPrior.cc

```diff
@@ -304,6 +305,19 @@ std::string CPrior::debug() const {
     return std::string();
 }
 
+CPrior::TStrStrPr CPrior::printMarginalLikelihoodStatistics() const {
+    TStrStrPr unknownValuePr{UNKNOWN_VALUE_STRING, UNKNOWN_VALUE_STRING};
+
+    if (this->isNonInformative()) {
+        return unknownValuePr;
+    }
+
+    try {
+        return this->doPrintMarginalLikelihoodStatistics();
+    } catch (...) {}
```

Comment:
I don't think it's good practice to use `catch (...)` except where it's effectively a `finally` clause and the caught exception is rethrown or otherwise propagated.

Reply:
Also, this method catches all its exceptions so I don't think we need a try here. (Unless you really can't avoid it, say callback into 3rd party code, none of our functions should be throwing exceptions.)